tests/regression/Cargo.toml

```diff
@@ -0,0 +1,15 @@
+[package]
+name = "regression"
+version = "0.1.0"
+edition = "2021"
+
+[dependencies]
+s2n-tls = { path = "../../bindings/rust/s2n-tls", features = ["testing"] }
+bytes = { version = "1", optional = true }
+errno = { version = "0.3" }
+libc = "0.2"
+crabgrind = "0.1"
+futures-test = "0.3.30"
+
+[profile.release]
+debug = true
```

Comment:
Does this work? I'd prefer to have this crate in the existing workspace.

Reply:
Yes, it works and enabling debug information is [necessary for Cachegrind](https://valgrind.org/docs/manual/cg-manual.html#:~:text=Cachegrind%20can%20annotate%20programs%20written%20in%20any%20language%2C%20so%20long%20as%20debug%20info%20is%20present%20to%20map%20machine%20code%20back%20to%20the%20original%20source%20code.%20Cachegrind%20has%20been%20used%20successfully%20on%20programs%20written%20in%20C%2C%20C%2B%2B%2C%20Rust%2C%20and%20assembly.) to accurately identify where instructions come from using the symbols. Setting debug in profile.release ensures that we retain the necessary debug information for detailed analysis while also tracking optimization in release. We can do that here or in the existing workspace but where the crate belongs is a different question.

Reply:
Sorry I should have been more specific. I thought you could only specify profile overrides at the workspace root, not individual crates in the workspace. If this isn't in the main workspace it needs a 
```toml
[workspace]
members = ["."]
```

Reply:
That being said... I would prefer that it is in the main workspace so these tests are run as part of the main test suite. Otherwise they get a bit neglected and people only find out they're broken in CI.

Reply:
That makes sense, it can run as part of the regular pre-PR workflow for devs once diffing is enabled by comparing local to the HEAD-1 commit. I'll make these changes in the next revision.

Reply:
In trying to add these tests to the existing workspace I ran into an issue where crabgrind can't compile due to an unresolved import 'std::ffi::c_char'. This type was [added to std](https://doc.rust-lang.org/stable/std/ffi/type.c_char.html) in Rust 1.64.0. Since the bindings/rust workspace requires the 1.63.0 rust-toolchain, this import doesn't resolve. We can either: 
1. Bump the rust-toolchain version: local development might miss breaking changes to the underlying Rust 1.63.0 bindings. However we would be able to include regression tests in the main test suites.
2. Include the regression test in a separate crate.

Let me know if you can think of any other paths forward.